### PR TITLE
Cleanup PV claim ref. during restore and not delete

### PR DESCRIPTION
**Pull request by ShyamsundarR**
Current code cleans up the PV claim ref. if metro DR is in use, at the time of deleting VRG on a cluster.

If the PVC is not deleted as part of the workload cleanup then due to the PV claim ref. being cleaned, the PVC reports its state as "Lost".

This is observed due to changes made as part of commit cf63370e3fd8025f1a55d64cc6880a3941b958fc which started supporting failover in cases where PVCs are not deleted.

The fix is to not perform the PV claim ref. cleanup on delete and instead continue to cleanup the PV claim ref. when the PV is restored to the cluster as part of validatePVExistence.

In validatePVExistence if the pvExists and matches the to be restored PV, and is Bound, it is ignored. If not, it is cleaned up subsequently by updateExistingPVForSync for metro DR cases. updateExistingPVForSync will ensure a recreated PVC will bind back to the PV, and pvExists would ensure that an already Bound PV is left as is.

Fixes: #652
Signed-off-by: Shyamsundar Ranganathan <srangana@redhat.com>

### CODE REVIEW

**Comment by ShyamsundarR** [comment]
e2e tested RDR with STS deployments and failover/relocate actions

File: controllers/vrg_volrep.go [hunk 1]
```diff
@@ -319,14 +319,13 @@ func (v *VRGInstance) preparePVCForVRDeletion(pvc *corev1.PersistentVolumeClaim,
...
+	// to the current cluster, in case the PVC has been deleted (cases like STS the
+	// PVC may not be deleted). This is achieved by clearing the required claim ref.
+	// such that the PV can bind back to a recreated PVC. func ref.: updateExistingPVForSync
 	if v.instance.Spec.ReplicationState == ramendrv1alpha1.Secondary &&
-		v.instance.Spec.Sync != nil {
-		if err := v.clearPVClaimRefMembers(pvc); err != nil {
-			return err
-		}
-	} else {
+		v.instance.Spec.Async != nil {
```
Comment (BenamarMk) [inline controllers/vrg_volrep.go:328]: (nit) Formatting

**Comment by BenamarMk** [comment]
> If the PVC is not deleted as part of the workload cleanup then due to the PV claim ref. being cleaned, the PVC reports its state as "Lost".

(Future) I think the VRG should take responsibility (from ACM or other custom operators like Crunchy) to own and delete all PVCs that it protects.  That will simplify things. In the case of MetroDR, we retain the PV as it is today. IOW, we should never leave PVCs behind even in the case of STS.

**Review by BenamarMk — APPROVED** [review]

**Review by raghavendra-talur — APPROVED** [review]

File: controllers/vrg_volrep.go [hunk 2]
```diff
@@ -319,14 +319,13 @@ func (v *VRGInstance) preparePVCForVRDeletion(pvc *corev1.PersistentVolumeClaim,
...
+	// Further, the PV will go to Available state, when the workload is moved back
+	// to the current cluster, in case the PVC has been deleted (cases like STS the
+	// PVC may not be deleted). This is achieved by clearing the required claim ref.
+	// such that the PV can bind back to a recreated PVC. func ref.: updateExistingPVForSync
 	if v.instance.Spec.ReplicationState == ramendrv1alpha1.Secondary &&
-		v.instance.Spec.Sync != nil {
-		if err := v.clearPVClaimRefMembers(pvc); err != nil {
-			return err
-		}
-	} else {
```
Comment (BenamarMk) [inline controllers/vrg_volrep.go:329]: How the heck did we miss this in the review? Go figure...
Removing the block above [325-329] causes the PV to not change the reclaim policy to `Delete` when the application is deleted.

**Pull request merged**